alpha/lib/model/LiveEntry.php

```diff
@@ -499,41 +499,133 @@ public function getScheduleEvents($startTime, $endTime)
+	public function isCurrentlyLive($currentDcOnly = false, $protocol = null)
+	{
+		return $this->getLiveStatus(true, $protocol) === EntryServerNodeStatus::PLAYABLE;
+	}
+
+	private function isExternalCurrentlyLive($reqProtocol = null)
 	{
-		if ($this->getViewMode() == ViewMode::PREVIEW)
+		$isLive = null;
+		$takeFirst = false;
+		$protocols = array();
+		switch ($reqProtocol)
 		{
-			if (!$this->canViewExplicitLive())
-				return false;
+			case KalturaPlaybackProtocol::HLS:
+			case KalturaPlaybackProtocol::APPLE_HTTP:
+				$protocols = array_unique(array($reqProtocol, KalturaPlaybackProtocol::HLS, KalturaPlaybackProtocol::APPLE_HTTP));
+				$takeFirst = true;
+				break;
+			case KalturaPlaybackProtocol::HDS:
+			case KalturaPlaybackProtocol::AKAMAI_HDS:
+				$protocols = array($reqProtocol);
+				break;
+
+			case null:
+				$configurations = $this->getLiveStreamConfigurations(requestUtils::getProtocol());
+				foreach ($configurations as $config)
+				{
+					$protocols[] = $config->getProtocol();
+				}
+				break;
 		}
 
-		if (kSimuliveUtils::getPlayableSimuliveEvent($this))
+		foreach($protocols as $protocol)
 		{
-			return true;
+			$isProtocolLive = $this->checkIsLiveByProtocol($protocol);
+			if ($isProtocolLive !== null)
+			{
+				if ($isProtocolLive || $takeFirst)
+				{
+					return $isProtocolLive;
+				}
+				$isLive = $isProtocolLive; // as false
+			}
 		}
 
-		$liveEntryServerNodes = $this->getPlayableEntryServerNodes();
-		if(!count($liveEntryServerNodes))
-			return false;
+		return $isLive;
+	}
 
-		/* @var LiveEntryServerNode $liveEntryServerNode*/
-		foreach($liveEntryServerNodes as $liveEntryServerNode)
+	private function checkIsLiveByProtocol($protocol) {
+		$config = $this->getLiveStreamConfigurationByProtocol($protocol, requestUtils::getProtocol());
+		if ($config)
 		{
-			/* @var WowzaMediaServerNode $serverNode*/
-			$serverNode = ServerNodePeer::retrieveActiveMediaServerNode(null, $liveEntryServerNode->getServerNodeId());
-			if($serverNode->getDc() == kDataCenterMgr::getCurrentDcId())
+			$url = $config->getUrl();
+			$backupUrl = $config->getBackupUrl();
+
+			$dpda = new DeliveryProfileDynamicAttributes();
+			$dpda->setEntryId($this->getId());
+			$dpda->setFormat($config->getProtocol());
+			KalturaLog::info("Determining status of live stream URL [ $url ] and Backup URL [ $backupUrl ]");
+
+			$urlManager = DeliveryProfilePeer::getLiveDeliveryProfileByHostName(parse_url($url, PHP_URL_HOST), $dpda);
+			if ($urlManager && $urlManager->isLive($url))
 			{
-				if ($this->getExplicitLive() && !$this->canViewExplicitLive() && !$liveEntryServerNode->getIsPlayableUser())
-					return false;
 				return true;
 			}
+			$urlManagerBackup = DeliveryProfilePeer::getLiveDeliveryProfileByHostName(parse_url($backupUrl, PHP_URL_HOST), $dpda);
+			return $urlManagerBackup ? $urlManagerBackup->isLive($backupUrl) : false;
 		}
-		
-		return !$currentDcOnly;
+		return null;
```

Comment:
if function is called checkIsLive, why not return true or false?

Reply:
NULL is mark the the protocol is not defined - because in that case we want to know when no protocol is set - the behavior is difference between protocol is defined but not live and when the requested protocol is not defined